plutus-ledger-api/test-onchain-evaluation/Main.hs

```diff
@@ -0,0 +1,121 @@
+{-# LANGUAGE QuasiQuotes      #-}
+{-# LANGUAGE RecordWildCards  #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Main (main) where
+
+import Codec.Serialise qualified as CBOR
+import Control.Concurrent.ParallelIO.Local qualified as Concurrent
+import Control.Exception (evaluate)
+import Data.ByteString.Base16 qualified as B16
+import Data.ByteString.Lazy qualified as Lazy
+import Data.Foldable (for_)
+import Data.List (intercalate)
+import Data.Maybe (catMaybes)
+import Data.Text (Text)
+import Data.Text.Encoding qualified as Text
+import Options.Applicative qualified as OA
+import Plutus.Script.Evaluation.Test.Options qualified as O
+import PlutusCore.Evaluation.Machine.CostModelInterface (CostModelApplyError)
+import PlutusLedgerApi.Test.EvaluationEvent (ScriptEvaluationData (..), ScriptEvaluationEvent (..),
+                                             ScriptEvaluationResult (..))
+import PlutusLedgerApi.V1 qualified as V1
+import PlutusLedgerApi.V2 qualified as V2
+import PyF (fmt)
+import System.Directory.Extra (listFiles)
+import System.Exit (die)
+import System.FilePath (isExtensionOf, (<.>))
+import System.IO.Extra (withTempDir)
+import System.Process.Extra (system_)
+
+data TestFailure
+    = InvalidEvaluationResult
+        Text
+        -- ^ Serialised `ScriptEvaluationEvent`
+        ScriptEvaluationResult
+    | InvalidCostParams
+        Text
+        -- ^ Serialised `ScriptEvaluationEvent`
+        CostModelApplyError
+    deriving stock (Show)
+
+main :: IO ()
```

Comment:
Can we separate the downloader and the test runner? Then we could make the latter a normal test suite (which means we get parallel test execution, failure reporting etc. for free). I imagine that locally I might want to run the test several times, and I don't want to be redoing the download every time! 

That would also make it easier to provide the data a different way if e.g. we want to make this work with Nix. I suggest that the test suite looks in an environment variable to find the data files.

Reply:
Could have a test case per block or epoch if necessary.

Reply:
Ok, fair point. At least "not clean up everything after testing" should be an option.

Reply:
`tasty` lets you define custom options if you define a new test type with `IsTest`!

Reply:
Ok, I removed the downloader from this PR.